encode.go

```diff
@@ -11,74 +11,92 @@ import (
 )
 
 // TODO: Handle all the errors
-func writeData(dbuf *bytes.Buffer, order binary.ByteOrder, data *K) error {
-	binary.Write(dbuf, order, data.Type)
-	if K0 <= data.Type && data.Type <= KT { // All Lists
-		binary.Write(dbuf, order, data.Attr)
-		binary.Write(dbuf, order, int32(reflect.ValueOf(data.Data).Len()))
-	} else if data.Type == XT { // Table
-		binary.Write(dbuf, order, data.Attr)
+func write(buf *bytes.Buffer, order binary.ByteOrder, k *K) error {
```

Comment:
Should we avoid change function/param names?

Reply:
Sure I will revert this